Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp

```diff
@@ -1411,6 +1412,35 @@ static void testWebViewIsWebProcessResponsive(WebViewTest* test, gconstpointer)
     g_assert_true(webkit_web_view_get_is_web_process_responsive(test->m_webView));
 }
 
+static void testWebViewIsWebProcessUnresponsive(WebViewTest* test, gconstpointer)
+{
+    static const char* hangHTML =
+        "<html>"
+        " <body>"
+        "  <script>"
+        "   setTimeout(function() {"
+        "    var start = new Date().getTime();"
+        "    var end = start;"
+        "     while(end < start + 14000) {"
+        "      end = new Date().getTime();"
+        "     }"
+        "    }, 500);"
+        "  </script>"
+        " </body>"
+        "</html>";
+
+    g_assert_true(webkit_web_view_get_is_web_process_responsive(test->m_webView));
+    test->loadHtml(hangHTML, nullptr);
+    test->waitUntilLoadFinished();
+    // Wait 1 second, so the js while loop kicks in and blocks the web process. Then restart the web process
+    // responsiveness timer. As the web process is busy, after 3 seconds the web process will be marked
```

Comment:
How does this work? Your new timer uses a 10 second timeout, so how can it detect unresponsiveness after just 3 seconds? It's not testing the original timer that you did not add, is it?

Even four seconds is an eternity in a big testsuite with hundreds of tests. How about adding a `webkitWebViewSetWebProcessResponsivenessTimeoutForTesting` to drop it from 10 seconds to a much lower value, like 500ms, for this test?

Reply:
ok, it seems to be unclear. When we restart the PeriodicalResponsivenessTimer, the first "beat" starts immediately and next is after 10 sec, and so on. I can change it and set a different interval (should it starts first "beat" after resetting the interval immediately or after the new interval?)

Reply:
Ah OK, I think it's fine as-is, then.

Reply:
(Um, but still do try to make the new test run faster. The original responsiveness test is already extremely slow.)

Reply:
To do this I would need to change the ResponsivenessTimer, if I change the interval of PeriodicalResponsivenessTimer I change only how often the ResponsivenessTimer will start, when it ends(3sec, then the notify::is-web-process-responsive is sent) is something separate.

Reply:
Ah, drat. OK, don't worry about it then. Our tests will just be 3-4 seconds slower than they were before.